### ISSUE TITLE
Error while running lein ring server on macOS Sierra

### ISSUE THREAD

**Issue by Varunram** [metabase/metabase#5249]
## Operating Systen: macOS Sierra 10.12.5

So, I was trying to setup the dev environment to try a few things out. I did `yarn` and `lein ring server` and I get the following error "Exception in thread "main" java.lang.AssertionError: Assert failed: Unable to connect to Metabase h2 DB". I tried the fixes suggested in similar threads, but they don't seem to work. I do have write permission to Downloads and I ran `sudo lein ring server` just to be sure. I tried copying the db to another location, not much luck with that either. Any suggestions as to where I might be going wrong? Thanks for your time!

The stack trace is below
```
Exception in thread "main" java.lang.AssertionError: Assert failed: Unable to connect to Metabase h2 DB.
(binding [*allow-potentailly-unsafe-connections* true] (require (quote metabase.driver)) ((resolve (quote metabase.driver/can-connect-with-details?)) engine details)), compiling:(/private/var/folders/29/2x_zy70x4ldc83743pss1dq40000gn/T/form-init7837298571320939294.clj:1:125)
	at clojure.lang.Compiler.load(Compiler.java:7391)
	at clojure.lang.Compiler.loadFile(Compiler.java:7317)
	at clojure.main$load_script.invokeStatic(main.clj:275)
	at clojure.main$init_opt.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invoke(main.clj:277)
	at clojure.main$initialize.invokeStatic(main.clj:308)
	at clojure.main$null_opt.invokeStatic(main.clj:342)
	at clojure.main$null_opt.invoke(main.clj:339)
	at clojure.main$main.invokeStatic(main.clj:421)
	at clojure.main$main.doInvoke(main.clj:384)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.AssertionError: Assert failed: Unable to connect to Metabase h2 DB.
(binding [*allow-potentailly-unsafe-connections* true] (require (quote metabase.driver)) ((resolve (quote metabase.driver/can-connect-with-details?)) engine details))
	at metabase.db$verify_db_connection.invokeStatic(db.clj:334)
	at metabase.db$verify_db_connection.invoke(db.clj:327)
	at metabase.db$verify_db_connection.invokeStatic(db.clj:330)
	at metabase.db$verify_db_connection.invoke(db.clj:327)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:382)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:376)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:103)
	at metabase.core$init_BANG_.invoke(core.clj:82)
	at clojure.lang.Var.invoke(Var.java:375)
	at ring.server.standalone$serve.invokeStatic(standalone.clj:95)
	at ring.server.standalone$serve.doInvoke(standalone.clj:75)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at ring.server.leiningen$serve.invokeStatic(leiningen.clj:13)
	at ring.server.leiningen$serve.invoke(leiningen.clj:10)
	at user$eval47160.invokeStatic(form-init7837298571320939294.clj:1)
	at user$eval47160.invoke(form-init7837298571320939294.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6927)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.load(Compiler.java:7379)
	... 14 more
```

**Comment by attekei** [comment]
@Varunram I have had exactly the same problem with my own development environment (macOS Sierra as well) and haven't found a solution for working with H2 yet.

I'm using PostgreSQL instead; the Homebrew version (`brew install postgresql`) works really well. You need the following environment variables for changing the application db:
```
MB_DB_TYPE=postgres MB_DB_HOST=localhost MB_DB_PORT=5432 MB_DB_USER=user MB_DB_DBNAME=metabase lein run
```
If you are only doing frontend changes, I recommend using `lein run` instead of `lein ring server` because of a faster server initialization. The only difference is that `lein run` doesn't watch and hot-reload changes in the Clojure code.

**Comment by Varunram** [comment]
@attekei Thanks a lot for your suggestions! I was about to test a few change on the front-end related to a few issues when I came across this. Postgres will do for now. 

> If you are only doing frontend changes, I recommend using lein run instead of lein ring server because of a faster server initialization

Thanks! Will try this out

**Comment by attekei** [comment]
One workaround for this based on my exploration today is to enforce the H2 server address to be `127.0.0.1`. I don't know what is the default IP address it tries to assign and why that assignment fails (or is that even the root cause).

If you are using the production jar:
```
java -Dh2.bindAddress=127.0.0.1 -jar metabase.jar
```
If you are in development environment (as in this issue):
```
LEIN_JVM_OPTS=-Dh2.bindAddress=127.0.0.1 lein run 
# or for hot reload:
LEIN_JVM_OPTS=-Dh2.bindAddress=127.0.0.1 lein ring server
```

**Comment by camsaul** [comment]
Fixed by #5396
